addons/website/static/src/components/views/page_list.js

```diff
@@ -37,20 +37,22 @@ export class PageListController extends PageControllerMixin(listView.Controller)
             actionMenuItems.other = actionMenuItems.other
                 .filter(item => !['archive', 'unarchive'].includes(item.key));
         }
-        actionMenuItems.other.splice(-1, 0, {
-            description: this.env._t("Publish"),
-            callback: async () => {
-                this.dialogService.add(ConfirmationDialog, {
-                    title: this.env._t("Publish Website Content"),
-                    body: sprintf(this.env._t("%s record(s) selected, are you sure you want to publish them all?"), this.model.root.selection.length),
-                    confirm: () => this.togglePublished(true),
-                });
-            }
-        },
-        {
-            description: this.env._t("Unpublish"),
-            callback: async () => this.togglePublished(false),
-        });
+        if (this.props.fields.hasOwnProperty('is_published')) {
```

Comment:
Note that I think `in` is more used, especially in the framework to do the same with `is_active`, but it's probably not changing anything.
(I was curious because my linter is not happy with `hasOwnProperty` for some reason)